File: scripts/seed_admin_user.py
#!/usr/bin/env python3
"""
Seed a temporary admin user into admin_platform.admin_users with a bcrypt-hashed password.
Usage (PowerShell):
  python scripts/seed_admin_user.py --nombre "Admin Temporal" --correo admin.temp@ventas.local --password "Admin2025!" --telefono 3000000000 --rol admin
Env overrides (optional): PGHOST, PGPORT, PGDATABASE, PGUSER, PGPASSWORD
"""
import os
import sys
import argparse
import bcrypt
import psycopg2
from psycopg2.extras import RealDictCursor


def get_conn():
    host = os.getenv('PGHOST', 'localhost')
    port = int(os.getenv('PGPORT', '5432'))
    db = os.getenv('PGDATABASE', 'sistemas_ventas')
    user = os.getenv('PGUSER', 'admin')
    pwd = os.getenv('PGPASSWORD', 'Ventas2025')
    return psycopg2.connect(host=host, port=port, dbname=db, user=user, password=pwd)


def main():
    ap = argparse.ArgumentParser()
    ap.add_argument('--nombre', required=True)
    ap.add_argument('--correo', required=True)
    ap.add_argument('--password', required=True)
    ap.add_argument('--telefono', default=None)
    ap.add_argument('--rol', default='admin', choices=['admin', 'soporte'])
    args = ap.parse_args()

    hashed = bcrypt.hashpw(args.password.encode('utf-8'), bcrypt.gensalt(rounds=10)).decode('utf-8')

    sql = """
    INSERT INTO admin_platform.admin_users (nombre, correo, telefono, contraseña, rol, activo)
    VALUES (%s, %s, %s, %s, %s, TRUE)
    ON CONFLICT (correo) DO UPDATE SET
      nombre = EXCLUDED.nombre,
      telefono = EXCLUDED.telefono,
      contraseña = EXCLUDED.contraseña,
      rol = EXCLUDED.rol,
      activo = TRUE,
      ultimo_ingreso = NULL
    RETURNING id, nombre, correo, rol, activo, fecha_creacion;
    """

    try:
        with get_conn() as conn:
            with conn.cursor(cursor_factory=RealDictCursor) as cur:
                cur.execute(sql, (args.nombre, args.correo, args.telefono, hashed, args.rol))
                row = cur.fetchone()
                print('Usuario seed OK:', dict(row))
        return 0
    except Exception as e:
        print('Error insertando usuario:', e, file=sys.stderr)
        return 1


if __name__ == '__main__':
    sys.exit(main())

File: src/scripts/seedAdminUser.js
import dotenv from 'dotenv';
dotenv.config();

import bcrypt from 'bcryptjs';
import { query } from '../config/db.js';

function parseArgs(argv) {
  const out = {};
  for (let i = 2; i < argv.length; i++) {
    const a = argv[i];
    if (a.startsWith('--')) {
      const [k, v] = a.replace(/^--/, '').split('=');
      if (typeof v === 'undefined') out[k] = true; else out[k] = v;
    }
  }
  return out;
}

async function main() {
  const args = parseArgs(process.argv);
  const nombre = args.nombre || 'Admin Temporal';
  const correo = args.correo || 'admin.temp@ventas.local';
  const telefono = args.telefono || null;
  const rol = args.rol || 'admin';
  const password = args.password || 'Admin2025!';
  const codigo = args.codigo || 'ADM-TEMP-0001';

  const hash = await bcrypt.hash(password, 10);
  // Detect if column 'codigo' exists
  const colCheck = await query(`
    SELECT 1 FROM information_schema.columns 
    WHERE table_schema='admin_platform' AND table_name='admin_users' AND column_name='codigo'
  `);
  let rows;
  if (colCheck.rowCount > 0) {
    const sql = `
      INSERT INTO admin_platform.admin_users (codigo, nombre, correo, telefono, contraseña, rol, activo)
      VALUES ($1,$2,$3,$4,$5,$6, TRUE)
      ON CONFLICT (correo) DO UPDATE SET
        codigo = EXCLUDED.codigo,
        nombre = EXCLUDED.nombre,
        telefono = EXCLUDED.telefono,
        contraseña = EXCLUDED.contraseña,
        rol = EXCLUDED.rol,
        activo = TRUE,
        ultimo_ingreso = NULL
      RETURNING id, codigo, nombre, correo, rol, activo, fecha_creacion;
    `;
    ({ rows } = await query(sql, [codigo, nombre, correo, telefono, hash, rol]));
  } else {
    const sql = `
      INSERT INTO admin_platform.admin_users (nombre, correo, telefono, contraseña, rol, activo)
      VALUES ($1,$2,$3,$4,$5, TRUE)
      ON CONFLICT (correo) DO UPDATE SET
        nombre = EXCLUDED.nombre,
        telefono = EXCLUDED.telefono,
        contraseña = EXCLUDED.contraseña,
        rol = EXCLUDED.rol,
        activo = TRUE,
        ultimo_ingreso = NULL
      RETURNING id, nombre, correo, rol, activo, fecha_creacion;
    `;
    ({ rows } = await query(sql, [nombre, correo, telefono, hash, rol]));
  }
  console.log('Usuario seed OK:', rows[0]);
}

main().catch((e) => {
  console.error('Error seed admin:', e);
  process.exit(1);
});
